qiskit_experiments/framework/experiment_data.py

```diff
@@ -2578,73 +2552,99 @@ def __str__(self):
         ret += f"\nFigures: {len(self._figures)}"
         return ret
 
+    def add_artifacts(self, artifacts: ArtifactData | list[ArtifactData], overwrite: bool = False):
+        """Add artifacts of experiment. The artifact ID must be unique.
 
-@contextlib.contextmanager
-def service_exception_to_warning():
-    """Convert an exception raised by experiment service to a warning."""
-    try:
-        yield
-    except Exception:  # pylint: disable=broad-except
-        LOG.warning("Experiment service operation failed: %s", traceback.format_exc())
-
+        Args:
+            artifacts: Artifact or list of artifacts to be added.
+            overwrite: Whether to overwrite the existing artifact.
+        """
+        if isinstance(artifacts, ArtifactData):
+            artifacts = [artifacts]
+
+        for artifact in artifacts:
+            if artifact.artifact_id in self._artifacts and not overwrite:
+                raise ValueError(
+                    "An artifact with id {artifact.id} already exists."
+                    "Set overwrite to True if you want to overwrite the existing"
+                    "artifact."
+                )
+            self._artifacts[artifact.artifact_id] = artifact
 
-class ExperimentStatus(enum.Enum):
-    """Class for experiment status enumerated type."""
+    def delete_artifact(
+        self,
+        artifact_key: int | str,
+    ) -> str | list[str]:
+        """Delete specified artifact data.
 
-    EMPTY = "experiment data is empty"
-    INITIALIZING = "experiment jobs are being initialized"
-    VALIDATING = "experiment jobs are validating"
-    QUEUED = "experiment jobs are queued"
-    RUNNING = "experiment jobs is actively running"
-    CANCELLED = "experiment jobs or analysis has been cancelled"
-    POST_PROCESSING = "experiment analysis is actively running"
-    DONE = "experiment jobs and analysis have successfully run"
-    ERROR = "experiment jobs or analysis incurred an error"
+        Args:
+            artifact_key: UID, name or index of the figure.
 
-    def __json_encode__(self):
-        return self.name
+        Returns:
+            Deleted artifact ids.
+        """
+        artifact_keys = self._find_artifact_keys(artifact_key)
 
-    @classmethod
-    def __json_decode__(cls, value):
-        return cls.__members__[value]  # pylint: disable=unsubscriptable-object
+        for key in artifact_keys:
+            del self._artifacts[key]
```

Comment:
There's no upstream method to delete files—it was probably not needed before. I'll open a separate issue for this.

Reply:
In fc058a9 I updated the behavior to overwrite an artifact file in the cloud with a size 0 file if all artifacts of that name have been deleted. This can be updated later to directly delete the artifact when that feature is added.